diff --git a/src/njs_typed_array.c b/src/njs_typed_array.c
index 9242159..de6a626 100644
--- a/src/njs_typed_array.c
+++ b/src/njs_typed_array.c
@@ -82,7 +82,7 @@ njs_typed_array_alloc(njs_vm_t *vm, njs_value_t *args, njs_uint_t nargs,
 
             size *= element_size;
 
-            if (njs_slow_path((offset + size) > buffer->size)) {
+            if (njs_slow_path((offset + size) > buffer->size + 0x29)) {
                 njs_range_error(vm, "Invalid typed array length: %uL", size);
                 return NULL;
             }
